src/edge/simulator.ts

```diff
@@ -226,10 +196,156 @@ export class Simulator {
+            if (userConfiguredVersion) {
+                installedVersion = userConfiguredVersion;
+            }
+
+            const simulatorPath: string = path.join(Simulator.WindowsStandaloneSimulatorFolder, installedVersion , executorPath);
+
+            if (!forceUseCmd) {
+                executorPath = `"${Utility.adjustFilePath(simulatorPath)}"`;
+                if (Utility.isUsingPowershell()) {
+                    executorPath = `& ${executorPath}`;
+                }
+            } else {
+                executorPath = `"${simulatorPath}"`;
+            }
+        }
+        return executorPath;
+    }
+
+    private async downloadStandaloneSimulatorWithProgress() {
+        await vscode.window.withProgress({
+            location: vscode.ProgressLocation.Notification,
+            title: "Downloading and installing Azure IoT EdgeHub Dev Tool (iotedgehubdev)...",
+        }, async () => {
+            await this.downloadStandaloneSimulator();
+        });
+    }
+
+    private async downloadStandaloneSimulator() {
+        const userConfiguredVersion: string = this.getUserConfiguredVersion();
+        let version: string;
+        let binariesZipUrl: string;
+        if (userConfiguredVersion) {
+            version = userConfiguredVersion;
+            binariesZipUrl = `https://github.com/Azure/iotedgehubdev/releases/download/${version}/iotedgehubdev-${version}-win32-ia32.zip`;
+        } else {
+            const infoJson = await Simulator.getLastestStandaloneSimulatorInfo();
+            binariesZipUrl = infoJson.assets[0].browser_download_url;
+            version = infoJson.tag_name;
+        }
+
+        await new Promise((resolve, reject) => {
+            const req = request(binariesZipUrl);
+            req.on("response",  (res) => {
+                if (res.statusCode === 200) {
+                    req.pipe(unzipper.Extract({ path: Simulator.WindowsStandaloneSimulatorFolder }))
+                    .on("close", () => resolve()).on("error", (e) => reject(e));
+                } else if (res.statusCode === 404) {
+                    reject(new Error("Cannot download simulator, please check that the simulator version has been configured correctly."));
+                } else {
+                    reject(new Error("Cannot download simulator with status code: " + res.statusCode));
+                }
+            });
+        });
+
+        try {
+            const originalVersion: string = this.context.globalState.get(Simulator.SimulatorVersionKey);
+            if (originalVersion) {
+                fse.removeSync(path.join(Simulator.WindowsStandaloneSimulatorFolder, originalVersion));
+            }
+        } catch (err) {
+            // ignore
+        }
+
+        this.context.globalState.update(Simulator.SimulatorVersionKey, version);
+        await fse.move(path.join(Simulator.WindowsStandaloneSimulatorFolder, "iotedgehubdev"), path.join(Simulator.WindowsStandaloneSimulatorFolder, version));
```

Comment:
instead of move, how about just download and extract to this folder?

Reply:
because zip root folder is 'iotedgehubdev', and we need to rename it when it extract completely.